CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/blaze/BlazeActions.java

```diff
@@ -82,24 +89,130 @@
+        }
+        BlazeActions.recentCustomColors.add(color);
+
+        // keep preferences up to date by storing new value whenever it is updated.
+        final StringBuffer recentColorsBuffer = new StringBuffer();
+        recentCustomColors.forEach(currentColor -> {
+            recentColorsBuffer.append(currentColor.getHtmlColor());
+            recentColorsBuffer.append(";");
+        });
+
+        // set all current colors into preferences.
+        prefs.put(GraphPreferenceKeys.BLAZE_RECENT_COLORS, recentColorsBuffer.toString());
+
+    }
+
+    /**
+     * Returns static list of colors used to generate recent blaze color menus.
+     *
+     * @return List of colors used to generate recent blaze color menus.
+     */
+    public static List<ConstellationColor> getRecentCustomColors() {
+        recentCustomColors.clear();
+        final String recentCustomColorsString = prefs.get(GraphPreferenceKeys.BLAZE_RECENT_COLORS,
+                GraphPreferenceKeys.BLAZE_RECENT_COLORS_DEFAULT);
+
+        for (final String currentColor : recentCustomColorsString.split(";")) {
+            if (!currentColor.equals("") && !currentColor.equals(" ")) {
+                // here the color should be populated correctly. Add to list.
+                recentCustomColors.add(ConstellationColor.fromHtmlColor(currentColor) == null
+                        ? ConstellationColor.getColorValue(currentColor)
+                        : ConstellationColor.fromHtmlColor(currentColor));
+            }
+        }
+
+        return recentCustomColors;
+    }
+
+    /**
+     * Returns static list of colors used to generate preset blaze color menus.
+     *
+     * @return List of colors used to generate preset blaze color menus.
+     */
+    public static List<ConstellationColor> getPresetCustomColors() {
+        presetCustomColors.clear();
+        final String presetCustomColorsString = prefs.get(GraphPreferenceKeys.BLAZE_PRESET_COLORS,
+                GraphPreferenceKeys.BLAZE_PRESET_COLORS_DEFAULT);
+
+        for (final String currentColor : presetCustomColorsString.split(";")) {
+            if (!currentColor.equals("") && !currentColor.equals(" ")) {
+                // here the color should be populated correctly. Add to list.
+                presetCustomColors.add(ConstellationColor.fromHtmlColor(currentColor) == null
+                        ? ConstellationColor.getColorValue(currentColor)
+                        : ConstellationColor.fromHtmlColor(currentColor));
+            }
+        }
+        return presetCustomColors;
+    }
+
+    /**
+     * Perform a deep copy of the buffered image instance to create a complete
+     * unique copy.
+     *
+     * @param bi The buffered image to copy.
+     * @return the copied image.
+     */
+    protected static BufferedImage copyImageBuffer(final BufferedImage bi) {
+        final ColorModel cm = bi.getColorModel();
+        final boolean isAlphaPremultiplied = cm.isAlphaPremultiplied();
```

Comment:
for the context it is being used in, I personally would just call `cm.isAlphaPreMultiplied()` when it is needed rather than storing it in a variable (the name of the variable seems intuitive from the function call)